### PR TITLE
Sisco/docker allow relative paths for invokeai data

### PR DESCRIPTION
**Pull request by dsisco11**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [X] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Community Node Submission

## Have you discussed this change with the InvokeAI team?
- [ ] Yes
- [X] No, because: it's small, discuss it in the comments

      
## Have you updated all relevant documentation?
- [ ] Yes
- [ ] No
- [X] Maybe?

## Description
This PR removes the docker-compose absolute path restriction for the data path of InvokeAI. 
The compose file will now bind the external Invoke data directory using the following envar values (in order of priority): 
1) `INVOKEAI_LOCAL_ROOT`
2) `INVOKEAI_ROOT`
3) `~/invokeai`

Since this change provides a fallback to the original `INVOKEAI_ROOT` variable, it should not impact existing users.

## Related Tickets & Documents

- Related Issue #
- Closes #

## Added/updated tests?

- [ ] Yes
- [X] No

### CODE REVIEW

**Review by ebr — COMMENTED** [review]
Thanks for the proposal. I _believe_ we had it like this at some point, but removed it because the usage was unclear, and it caused issues for those who ran invoke both in and outside of docker.

Perhaps we could solve this by introducing a new env var `CONTAINER_INVOKEAI_ROOT`, defaulting to `/invokeai`? This way behaviour of `INVOKEAI_ROOT` remains the same between bare-metal and docker setups, and the new env var can be used if there's a need to change the root path in container. 
something like:
```
source: ${INVOKEAI_ROOT:-~/invokeai}
target: ${CONTAINER_INVOKEAI_ROOT:-/invokeai}
```

Would that work - thoughts?

**Comment by dsisco11** [comment]
But wait, isn't the `INVOKEAI_ROOT` envar used internally by various pieces of InvokeAIs backend code so it can find the root dir? So wouldn't having the `INVOKEAI_ROOT` var be different to the folder that the container internally binds prevent invoke from finding the dir correctly?

**Comment by ebr** [comment]
Sounds good. May I ask that we name it `HOST_INVOKEAI_ROOT`? that way we make it clear that it's the invoke root on the docker host rather than in container. Otherwise this hits all the marks as you've described. Please rename and I'll test and approve. Thank you!

**Comment by dsisco11** [comment]
done!

File: docker/.env.sample [hunk 1]
```diff
@@ -2,10 +2,13 @@
 ## Any environment variables supported by InvokeAI can be specified here,
 ## in addition to the examples below.
 
-# INVOKEAI_ROOT is the path to a path on the local filesystem where InvokeAI will store data.
+# HOST_INVOKEAI_ROOT is the path on the docker hosts filesystem where InvokeAI will store data.
 # Outputs will also be stored here by default.
-# This **must** be an absolute path.
-INVOKEAI_ROOT=
+# If relative, it will be relative to the docker directory in which the docker-compose.yml file is located
+HOST_INVOKEAI_ROOT=../../invokeai-data
```
Comment (ebr) [inline docker/.env.sample:8]: we don't want this to be the default for users who copy .env verbatim
```suggestion
#HOST_INVOKEAI_ROOT=../../invokeai-data
```

**Review by ebr — APPROVED** [review]

**Review by hipsterusername — APPROVED** [review]

**Pull request merged**